### PR TITLE
sync restart

### PR DESCRIPTION
**Pull request by totto82**
Stop reading the restart values twice.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
@joakim-hove Is the summaryState sufficiently tested such that I can trust Jenkins that this PR doesn't mess up something. 
@atgeirr I have verified that restart in flow where next time step is passed using EXTRA works.

**Comment by totto82** [comment]
@joakim-hove @atgeirr ping!

**Comment by atgeirr** [comment]
Should be merged. I'll rerun jenkins since it was run a while ago.

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by blattms** [comment]
Nice, need to update #2033 afterwards.

**Comment by atgeirr** [comment]
> Nice, need to update #2033 afterwards.

I assume that means you are ok with merging this first?

**Comment by blattms** [comment]
I basically do not care what comes in first

**Comment by atgeirr** [comment]
> I basically do not care what comes in first

I'll merge this now then.

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -493,6 +493,8 @@ namespace Opm {
 
         const Wells* wells = wellsmanager.c_wells();
 
+        wells_ecl_ = schedule().getWells2(report_step);
```
Comment (blattms) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp]: I really need to find out why this PR broke my branch.
Can you please tell me why this is needed here? Previously, this was only done in beginReportStep and now additionally here in initFromRestartFile

Comment (atgeirr) [reply opm/simulators/wells/BlackoilWellModel_impl.hpp]: I do not see any obvious flaw here unfortunately. What is the broken behaviour?

Comment (totto82) [reply opm/simulators/wells/BlackoilWellModel_impl.hpp]: Before beginReportStep was called in the restart process. Now it is not. Maybe adding the following lines to initFromRestartFile helps 

```
wells_active_ = localWellsActive() ? 1 : 0;
wells_active_ = grid.comm().max(wells_active_);
```

**Comment by blattms** [comment]
Makes sense in the release and will there save me from a merged conflict.

**Comment by blattms** [comment]
backported to release via #2087

**Pull request merged**